src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+
+const useFetchPostNotificationHook = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+
+  const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
+    if (!localDb) return;
+    const allPromises: Promise<void>[] = notifications.map((notification) => {
+      const channelList = fromNotificationAPI(notification);
+      return channelList.saveIfLatest(localDb);
+    });
+
+    try {
+      await Promise.all(allPromises);
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving notifications:', e);
+    }
+  };
+
+  const getAllSignedPostNotifications = async () => {
+    try {
+      const signedPostNotifications = await SignedMessageRepo.getAllSignedPostNotifications();
+      saveNotifications(signedPostNotifications, ChannelList.fromSignedPostNotificationAPI);
+    } catch (e) {
+      console.log('error on getting signedPostNotifications:', e);
+    }
+  };
+
+  const getAllAnonymousPostNotifications = async () => {
+    try {
+      const anonymousPostNotifications =
+        await AnonymousMessageRepo.getAllAnonymousPostNotifications();
+      saveNotifications(anonymousPostNotifications, ChannelList.fromAnonymousPostNotificationAPI);
+    } catch (e) {
+      console.log('error on getting anonymousPostNotifications', e);
+    }
+  };
+
+  return {
+    getAllSignedPostNotifications,
+    getAllAnonymousPostNotifications
+  };
+};
+
+export default useFetchPostNotificationHook;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this code is done by logging the error to the console. This might not be sufficient for a production environment. Consider using a more robust error handling strategy, such as sending these errors to an error tracking service or displaying user-friendly error messages.

```diff
-     console.log('error on saving notifications:', e);
+     // Replace with your error handling logic
+     handleError(e);

-     console.log('error on getting signedPostNotifications:', e);
+     // Replace with your error handling logic
+     handleError(e);

-     console.log('error on getting anonymousPostNotifications', e);
+     // Replace with your error handling logic
+     handleError(e);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->